File: src/main/java/com/github/svegon/capi/event/render/EntityHasOutlineCallback.java
package com.github.svegon.capi.event.render;

import net.fabricmc.fabric.api.event.Event;
import net.fabricmc.fabric.api.event.EventFactory;
import net.minecraft.entity.Entity;
import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;

public interface EntityHasOutlineCallback {
    Event<EntityHasOutlineCallback> EVENT = EventFactory.createArrayBacked(EntityHasOutlineCallback.class,
            (entity, callback) -> {}, listeners -> (entity, callback) -> {
        for (EntityHasOutlineCallback listener : listeners) {
            listener.hasEntityOutline(entity, callback);

            if (callback.isCancelled()) {
                return;
            }
        }
            });

    void hasEntityOutline(Entity entity, CallbackInfoReturnable<Boolean> callback);
}

File: src/main/kotlin/io/github/svegon/mclientapi/event/render/EntityHasOutlineCallback.kt
package io.github.svegon.capi.event.render

import io.github.svegon.capi.event.render.EntityHasOutlineCallback
import net.fabricmc.fabric.api.event.Event
import net.fabricmc.fabric.api.event.EventFactory
import net.minecraft.entity.Entity
import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable
import java.util.function.Function
import kotlin.Array
import kotlin.Boolean
import kotlin.invoke

interface EntityHasOutlineCallback {
    fun hasEntityOutline(entity: Entity?, callback: CallbackInfoReturnable<Boolean?>?)

    companion object {
        @JvmField
        val EVENT: Event<EntityHasOutlineCallback?> = EventFactory.createArrayBacked(
            EntityHasOutlineCallback::class.java,
            EntityHasOutlineCallback { entity: Entity?, callback: CallbackInfoReturnable<Boolean?>? -> },
            Function { listeners: Array<EntityHasOutlineCallback?>? ->
                EntityHasOutlineCallback { entity: Entity?, callback: CallbackInfoReturnable<Boolean?>? ->
                    for (listener in listeners) {
                        listener.hasEntityOutline(entity, callback)

                        if (callback.isCancelled()) {
                            return@EntityHasOutlineCallback
                        }
                    }
                }
            })
    }
}
